src/flask_restalchemy/tests/test_query_callback.py

```diff
@@ -0,0 +1,118 @@
+import pytest
+from flask import json, request
+from flask_restalchemy.serialization import ModelSerializer, Field, NestedModelField
+
+from flask_restalchemy import Api
+from flask_restalchemy.tests.sample_model import Employee, Company, Department, Address, employee_department, db
+
+
+class EmployeeSerializer(ModelSerializer):
+
+    password = Field(load_only=True)
+    created_at = Field(dump_only=True)
+    company_name = Field(dump_only=True)
+    address = NestedModelField(Address)
+
+@pytest.fixture(autouse=True)
+def create_test_sample(db_session):
+    protoss = Company(id=1, name='Protoss')
+    zerg = Company(id=2, name='Zerg')
+    terrans = Company(id=3, name='Terrans')
+
+    basic = Department(id=100, name='Basic')
+    heroes = Department(id=200, name='Heroes')
+
+    raynor = Employee(id=11, firstname='Jim', lastname='Raynor', company=terrans, departments=[basic, heroes])
+    marine = Employee(id=12, firstname='John', lastname='Doe', company=terrans, departments=[basic])
+    kerrigan = Employee(id=13, firstname='Sarah', lastname='Kerrigan', company=terrans, departments=[heroes])
+
+    tassadar = Employee(id=21, firstname='Tassadar', lastname='The Warrior', company=protoss, departments=[heroes])
+    zealot = Employee(id=22, firstname='Aiur', lastname='Zealot', company=protoss, departments=[basic])
+
+    zergling = Employee(id=31, firstname='Rush', lastname='Lings', company=zerg, departments=[basic])
+
+    db_session.add_all([protoss, zerg, terrans])
+    db_session.add_all([basic, heroes])
+    db_session.add_all([raynor, kerrigan, marine, tassadar, zealot, zergling])
+    db_session.commit()
+
+@pytest.fixture(autouse=True)
+def sample_api(flask_app):
+    api = Api(flask_app)
```

Comment:
Move Api object creation to tests and add only the query_modifier you are testing